core/src/test/scala/org/apache/spark/status/AppStatusStoreSuite.scala

```diff
@@ -132,10 +170,49 @@ class AppStatusStoreSuite extends SparkFunSuite {
   }
 
   private def newTaskData(i: Int, status: String = "SUCCESS"): TaskDataWrapper = {
-    new TaskDataWrapper(
-      i, i, i, i, i, i, i.toString, i.toString, status, i.toString, false, Nil, None,
-      i, i, i, i, i, i, i, i, i, i,
+
+    val metrics = new v1.TaskMetrics(
       i, i, i, i, i, i, i, i, i, i,
-      i, i, i, i, stageId, attemptId)
+      new InputMetrics(i, i),
+      new OutputMetrics(i, i),
+      new ShuffleReadMetrics(i, i, i, i, i, i, i),
+      new ShuffleWriteMetrics(i, i, i))
+
+    val hasMetrics = i >= 0
+
+    val taskMetrics: v1.TaskMetrics = if (hasMetrics && status != "SUCCESS") {
```

Comment:
How about using `LiveTask` here instead of repeating a bunch of that code?

Reply:
Yes, now using LiveTask.